Sources/StreamChat/ChatClient.swift

```diff
@@ -49,206 +49,48 @@ public class ChatClient {
     private(set) var connectionRecoveryHandler: ConnectionRecoveryHandler?
 
     /// The notification center used to send and receive notifications about incoming events.
-    private(set) lazy var eventNotificationCenter: EventNotificationCenter = {
-        let center = environment.notificationCenterBuilder(databaseContainer)
+    private(set) var eventNotificationCenter: EventNotificationCenter
 
-        let middlewares: [EventMiddleware] = [
-            EventDataProcessorMiddleware(),
-            TypingStartCleanupMiddleware(
-                excludedUserIds: { [weak self] in Set([self?.currentUserId].compactMap { $0 }) },
-                emitEvent: { [weak center] in center?.process($0) }
-            ),
-            ChannelReadUpdaterMiddleware(
-                newProcessedMessageIds: { [weak center] in center?.newMessageIds ?? [] }
-            ),
-            UserTypingStateUpdaterMiddleware(),
-            ChannelTruncatedEventMiddleware(),
-            MemberEventMiddleware(),
-            UserChannelBanEventsMiddleware(),
-            UserWatchingEventMiddleware(),
-            UserUpdateMiddleware(),
-            ChannelVisibilityEventMiddleware(),
-            EventDTOConverterMiddleware()
-        ]
+    // MARK: Repositories
 
-        center.add(middlewares: middlewares)
+    private(set) var connectionRepository: ConnectionRepository
 
-        return center
-    }()
-
-    // MARK: Repositories
+    private(set) var authenticationRepository: AuthenticationRepository
```

Comment:
Will these change at some point? Maybe `let` would be better?

Reply:
Yes this can be all `let` 👍 They were `var` becasue they were lazy before, now we can change them to `let`. Will do 👍